elasticdl/python/common/k8s_volume.py

```diff
@@ -16,32 +22,54 @@ def parse_volume_and_mount(volume_conf, pod_name):
         volumes (List): a Python list contains k8s volumes.
         volume_mounts (List): a Python list contains k8s volume mounts.
     """
-    volumes = []
+    volumes_cache = {}
```

Comment:
Why is this called cache?

Reply:
It's used to store the built volume while parsing the volume configuration string. We use it to avoid creating the volume with the same name and type.
Maybe renaming it volumes_per_type will be more suitable.